src/field.rs

```diff
@@ -123,20 +183,37 @@ macro_rules! make_field {
         pub struct $elem(u128);
 
         impl $elem {
-            fn try_from_bytes(bytes: &[u8], mask: u128) -> Result<Self, FieldError> {
+            /// Attempts to instantiate an `$elem` from the first `Self::BYTES` little-endian bytes
+            /// in the provided slice. If `mask` is `Some`, it will be applied to decoded outputs
+            /// before checking them against the modulus.
+            ///
+            /// # Errors
+            ///
+            /// An error is returned if the provided slice is not long enough to encode a field
+            /// element or if the decoded value is greater than the field prime.
+            ///
+            /// # Notes
+            ///
+            /// We cannot use `u128::from_le_bytes` here because that function expects inputs
+            /// to be exactly 16 bytes long. Our encoding of most field elements is more compact,
+            /// and does not always correspond to the size of an integer type (specifically,
+            /// `Field80`'s encoding is 10 bytes, even though it is a 16 byte `u128` in memory).
+            fn try_from_bytes(bytes: &[u8], mask: Option<u128>) -> Result<Self, FieldError> {
```

Comment:
Why let `mask` be optional? I ask because branch prediction is probably more expensive than bitwise-AND.

Reply:
You're suggesting that `try_from_reader` could just do `try_from_bytes(&bytes, u128::MAX)`? I think I do like that better than this version!